include/kokkos-utils/callbacks/ConjunctionMatcher.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef KOKKOS_UTILS_CALLBACKS_CONJUNCTIONMATCHER_HPP
+#define KOKKOS_UTILS_CALLBACKS_CONJUNCTIONMATCHER_HPP
+
+#include "kokkos-utils/callbacks/Events.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+/**
+ * @brief Conjunction of matchers that is @c true only if all @ref matchers agree.
+ *
+ * @todo Try to improve usability. For now, the construction requires a @c std::tuple,
+ *       and it would not be an aggregate if we add a user-defined constructor.
+ */
+template <typename... Matchers>
+struct ConjunctionMatcher
+{
+    template <Event EventType>
+    bool operator()(const EventType& event)
```

Comment:
Just wondering if we should write a requires clause to constrain the EventType based on the event types that the child matchers can match? Probably good to write a test where one of the child matchers matches only a subset of the events, and then use the `Matcher` concept to see whether that subset gets propagated to the `ConjunctionMatcher`.

Reply:
It could be done, but I guess it's too heavy for the compiler.

Reply:
OK, it's done !

It was indeed necessary to constrain the call operator of `ConjunctionMatcher` to only accept events that are accepted by all sub-matchers, otherwise it was reporting as a matcher for *any event but would fail compiling.